transporters/src/main/java/com/nextdoor/bender/ipc/splunk/SplunkTransportConfig.java

```diff
@@ -0,0 +1,186 @@
+ *
+ * Copyright 2017 Nextdoor.com, Inc
+ *
+ */
+
+package com.nextdoor.bender.ipc.splunk;
+
+import java.io.UnsupportedEncodingException;
+
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaDefault;
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaDescription;
+import com.nextdoor.bender.ipc.TransportConfig;
+import com.nextdoor.bender.utils.Passwords;
+
+@JsonTypeName("Splunk")
+@JsonSchemaDescription("Writes events to a Splunk HEC endpoint.")
+public class SplunkTransportConfig extends TransportConfig {
+
+  @JsonSchemaDescription("Splunk HTTP endpoint hostname.")
+  @JsonProperty(required = true)
+  private String hostname = null;
+
+  @JsonSchemaDescription("Splunk HTTP endpoint port.")
+  @JsonSchemaDefault(value = "443")
+  @JsonProperty(required = false)
+  @Min(1)
+  @Max(65535)
+  private Integer port = 443;
+
+  @JsonSchemaDescription("Splunk auth token.")
+  @JsonProperty(required = true)
+  private String authToken = null;
+
+  @JsonSchemaDescription("Splunk data index.")
+  @JsonProperty(required = true)
+  private String index = null;
+
+  @JsonSchemaDescription("Use SSL connections (certificates are not validated).")
+  @JsonSchemaDefault(value = "false")
+  @JsonProperty(required = false)
+  private Boolean useSSL = false;
+
+  @JsonSchemaDescription("Use GZIP compression on HTTP calls.")
+  @JsonSchemaDefault(value = "false")
+  @JsonProperty(required = false)
+  private Boolean useGzip = false;
+
+  @JsonSchemaDescription("Maximum number of documents in  bulk api call.")
+  @JsonSchemaDefault(value = "500")
+  @JsonProperty(required = false)
+  @Min(500)
+  @Max(100000)
+  private Integer batchSize = 500;
+
+  @JsonSchemaDescription("Number of retries to make when a put failure occurs.")
+  @JsonSchemaDefault(value = "0")
+  @JsonProperty(required = false)
+  @Min(0)
+  @Max(10)
+  private Integer retryCount = 0;
+
+  @JsonSchemaDescription("Initial delay between retries. If more than one retries specified exponential backoff is used.")
+  @JsonSchemaDefault(value = "1000")
+  @JsonProperty(required = false)
+  @Min(1)
+  @Max(60000)
+  private Long retryDelay = 1000l;
+
+  @JsonSchemaDescription("Socket timeout on HTTP connection.")
+  @JsonSchemaDefault(value = "40000")
+  @JsonProperty(required = false)
+  @Min(1000)
+  @Max(300000)
+  private Integer timeout = 40000;
+
+  public String getHostname() {
+    return hostname;
+  }
+
+  public void setHostname(String hostname) {
+    this.hostname = hostname;
+  }
+
+  public Integer getPort() {
+    return port;
+  }
+
+  public void setPort(Integer port) {
+    this.port = port;
+  }
+
+  public void setAuthToken(String authToken) {
+    this.authToken = authToken;
+  }
+
+  public String getAuthToken() {
+    /*
+     * If token uses KMS then decrypt it.
```

Comment:
This comment is a bit confusing - does the `Password.getPassword(this.authToken)` take care of this? If so, I feel like this comment should be reworked to something like: `Get the Auth token... decryption happens automatically if necessary`.